json_tests/tests/test_json.rs

```diff
@@ -1723,3 +1737,26 @@ fn test_stack_overflow() {
         (&brackets, Error::Syntax(ErrorCode::Custom("recursion limit exceeded".into()), 1, 128)),
     ]);
 }
+
+#[test]
+fn test_allow_ser_integers_as_map_keys(){
+    let map = treemap!(
+        1 => 2,
+        2 => 4,
+        -1 => 6,
+        -2 => 8
+    );
+    
+    serde_json::to_vec(&map).unwrap();
```

Comment:
Please test the actual JSON string that it produces. You might be surprised what it is right now:

```json
{"\""-2"\"":8,"\""-1"\"":6,"\""1"\"":2,"\""2"\"":4}
```